### ISSUE TITLE
Passwords are viewable as plaintext in qutebrowser's debug log and might be shared if you decide to submit a crash report

### ISSUE THREAD

**Issue by gerrywastaken** [qutebrowser/qutebrowser#6589]
Is this still the current situation?

> WARNING: The login details are viewable as plaintext in qutebrowser's debug log (qute://log) and might be shared if
you decide to submit a crash report!"""
https://github.com/qutebrowser/qutebrowser/blob/master/misc/userscripts/qute-pass#L49-L50

> WARNING: the passwords are stored in qutebrowser's debug log reachable via the url qute://log
https://github.com/qutebrowser/qutebrowser/blob/master/misc/userscripts/password_fill#L10-L13

I can see an issue meant to address this, but it [was closed](https://github.com/qutebrowser/qutebrowser/pull/5289) in favour of another issue, that was apparently fixed by [this PR](https://github.com/qutebrowser/qutebrowser/pull/5445). Is the solution to set logging to critical for in-memory and on disk logs, if you are using a password manager?

```python
+# Level for in-memory logs.
+# Type: LogLevel
+# Valid values:
+#   - vdebug
+#   - debug
+#   - info
+#   - warning
+#   - error
+#   - critical
+c.logging.level.ram = 'critical'
+
+# Level for console (stdout/stderr) logs. Ignored if the `--loglevel` or
+# `--debug` CLI flags are used.
+# Type: LogLevel
+# Valid values:
+#   - vdebug
+#   - debug
+#   - info
+#   - warning
+#   - error
+#   - critical
+c.logging.level.console = 'critical'
```

If so, can the advice be added to a page somewhere and linked from the warnings in the userscripts? Otherwise this essentially nullifies any security provided by their password manager.

**Comment by The-Compiler** [comment]
Yes, that's indeed still the current situation. It probably won't change until we have a proper way to fill in passwords as part of #30. I don't think a dedicated page is needed, but pointing out that `logging.level.ram` can be set to `info` to prevent this as part of the warning would indeed be good. Would you like to open a PR which does that?

(FWIW, no need to set it to `critical`, `info` should suffice)

**Comment by gerrywastaken** [comment]
I still haven't had time to look into this, but I will probably have some time in a few days. Thanks for letting me know.

**Comment by gerrywastaken** [comment]
Just as an update. I'm actually running into similar issues to what you encountered here:
https://github.com/qutebrowser/qutebrowser/issues/384#issuecomment-92300232

This is when I'm trying to run one one of the password userscripts
```
Process stderr:
objc[44109]: Class QMacAutoReleasePoolTracker is implemented in both /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyQt5/Qt5/lib/QtCore.framework/Versions/5/QtCore (0x104749198) and /Applications/qutebrowser.app/Contents/MacOS/QtCore (0x11f5c5198). One of the two will be used. Which one is undefined.
objc[44109]: Class QT_ROOT_LEVEL_POOL__THESE_OBJECTS_WILL_BE_RELEASED_WHEN_QAPP_GOES_OUT_OF_SCOPE is implemented in both /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyQt5/Qt5/lib/QtCore.framework/Versions/5/QtCore (0x104749210) and /Applications/qutebrowser.app/Contents/MacOS/QtCore (0x11f5c5210). One of the two will be used. Which one is undefined.
objc[44109]: Class KeyValueObserver is implemented in both /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyQt5/Qt5/lib/QtCore.framework/Versions/5/QtCore (0x104749238) and /Applications/qutebrowser.app/Contents/MacOS/QtCore (0x11f5c5238). One of the two will be used. Which one is undefined.
objc[44109]: Class RunLoopModeTracker is implemented in both /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyQt5/Qt5/lib/QtCore.framework/Versions/5/QtCore (0x104749288) and /Applications/qutebrowser.app/Contents/MacOS/QtCore (0x11f5c5288). One of the two will be used. Which one is undefined.
QObject::moveToThread: Current thread (0x7ff22ec35d80) is not the object's thread (0x7ff23110e590).
Cannot move to target thread (0x7ff22ec35d80)

You might be loading two sets of Qt binaries into the same process. Check that all plugins are compiled against the right Qt binaries. Export DYLD_PRINT_LIBRARIES=1 and check that only one set of binaries are being loaded.
qt.qpa.plugin: Could not load the Qt platform plugin "cocoa" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: cocoa, minimal, offscreen, webgl.
```

Looking at possible fixes to this is leading me down several rabbit holes, however I want to test any assertions I'm making when updating the docs, especially as they have an impact on possible password compromise.

**Comment by The-Compiler** [comment]
The first couple of lines (the `objc` ones) are likely harmless and unrelated. The problem is this:

> You might be loading two sets of Qt binaries into the same process. Check that all plugins are compiled against the right Qt binaries. Export DYLD_PRINT_LIBRARIES=1 and check that only one set of binaries are being loaded.

and especially:

> qt.qpa.plugin: Could not load the Qt platform plugin "cocoa" in "" even though it was found.

I'm not sure what's going on there, as userscripts run in an external process separate from qutebrowser - but it really seems like an issue with your system/setup rather than qutebrowser itself. If you open a PR I can give those scripts a quick try on Linux, but I'm pretty confident nothing sensitive is going to be logged with the default info level.

**Comment by gerrywastaken** [comment]
> but it really seems like an issue with your system/setup rather than qutebrowser itself

Perhaps, but as mentioned the script runs fine outside of qutebrowser.

> but I'm pretty confident nothing sensitive is going to be logged with the default info level.

Yes as you have deep understanding of qutebrowser and how it works it does seem like this would just be a small PR for a text change. But for me the shortest path would be vouching for your claim. As it's related to security of a system I don't fully understand and reducing an existing warning somebody put there for a reason; I'm required to do dude diligence and check each of these user scripts and the impact of log changes. I should also check through the logs for each commit where that warning was added to make sure the commit message doesn't contain additional information as to scenarios I'm not aware of. Which is made more complicated by the errors I'm encountering.

In general I'd say the easiest thing would be for a person making a security claim to be the one committing it. I understand that you want to encourage more community involvement and PRs, but I'm not sure me being the one making *this* PR is a good use of anybody's time.
